### DISCUSSION TITLE
A question about how to record a phase space of a particular size

### DISCUSSION THREAD

**Discussion by CQUswl** [nrc-cnrc/EGSnrc#1218] [General]
I wanted to record the proportion of electrons scattered to different positions after the electrons were incident on the target, so I wrote an EGS++ program.I set up two phase space recording positions.‘’scoreplane1‘’ is perpendicular to the Z-axis. ‘’scoreplane2‘’ is perpendicular to the Y axis.Is there any way to set scoreplane1 and scoreplane1 to a specific location, a specific size of the recording surface. For example, a recording surface with a size of 1 mm × 1 mm perpendicular to the Y-axis.

################################
#
# Simple slab simulation
#
################################
################################
################################
:start run control:
ncase = 1e4 # The number of histories to simulate
:stop run control:
################################
################################
:start MC transport parameter:
    Global ECUT                    = 0.512          # Global electron transport cutoff
    Global PCUT                    = 0.001          # Global photon transport cutoff
    Global SMAX                    = 1e10           # Global maximum step-size restriction for e- transport
    ESTEPE                         = 0.25           # Default is 0.25
    XIMAX                          = 0.5            # Default is 0.5, max. value is 1.
    Boundary crossing algorithm    = exact          # exact (default), PRESTA-I
    Skin depth for BCA             = 3              # Default value is 3 for exact boundary crossing
    Electron-step algorithm        = PRESTA-II      # PRESTA-II (default),PRESTA-I
    Spin effects                   = On             # On (default),Off
    Brems angular sampling         = KM             # Simple,KM (default)
    Brems cross sections           = NIST             # BH (default), NIST, NRC
    Pair angular sampling          = Simple         # Off, Simple (default),KM
    Triplet production             = Off            # On or Off (default)
    Electron Impact Ionization     = On             # On, Off (default), casnati, kolbenstvedt, gryzinski
    Bound Compton scattering       = Simple             # On, Off, Simple, norej (default)
    Radiative Compton corrections  = On            # On,  Off (default)
    Rayleigh scattering            = On             # On ,Off (default), custom
    Photoelectron angular sampling = On             # On (default),Off
    Atomic relaxations             = On             # On (default),Off
    Photon cross sections          = xcom           # si, epdl, xcom (default) or user-supplied
    Photon cross-sections output   = Off            # Off (default) or On
    Photonuclear attenuation       = On            # Off (default) or On
    Photonuclear cross sections    = default        # default (default) or user-supplied
:stop MC transport parameter:
################################
################################
:start geometry definition: # Only 1 geometry definition block allowed
### Define a plate of tantalum
:start geometry:
name     = air_box
library  = egs_box
box size = 25 25 50
:start media input:
media = vacuum
:stop media input:
:stop geometry:
:start geometry:
name        = scoreplane1
library     = egs_planes
type        = EGS_Zplanes
positions   = -0.1
:stop geometry:
:start geometry: # Many geometry blocks can be defined
name = scoreplane2 # This name can be anything you like
library = egs_planes
type = EGS_Yplanes
positions   = -1.1
:stop geometry:
:start geometry:
library  = egs_planes
type     = EGS_Planes
name     = planes
normal   = 0 0.1228 1
positions = 0        
:stop geometry:
:start geometry: # Many geometry blocks can be defined
name = slab # This name can be anything you like
library = egs_ndgeometry
type = EGS_XYZGeometry
x-planes = -1.5, 1.5 # in cm
y-planes = -1, 1.5 # in cm
z-planes = 0, 2.5# in cm
:stop geometry:
:start geometry:
library = egs_ndgeometry
name    = target
dimensions = planes slab
:start media input:
media = tungsten # Indexed as 0 1
# Other regions default to medium 0 (vacuum)
:stop media input:
:stop geometry:
:start geometry:
name    = target_in_vacuum
library = egs_genvelope
base geometry = air_box
inscribed geometries = target
:stop geometry:
### Use the geometry with this name for the simulation
simulation geometry = target_in_vacuum
:stop geometry definition:
################################
################################
:start media definition: # Only 1 media definition block allowed
# Defining media in the input file like this is called "pegsless" mode
### Cutoff energies, in MeV
ae = 0.512 # lowest energy for electrons (kinetic+0.511)
ap = 0.001 # lowest energy for photons (kinetic)
ue = 50.511 # maximum energy for electrons (kinetic+0.511)
up = 50 # maximum energy for photons (kinetic)
### tungsten
:start tungsten: # this name can be anything
density correction file = tungsten # name the file (no ext.)
:stop tungsten:
:stop media definition:
################################
################################
:start source definition: # Only 1 source definition block allowed
### Pencil beam
:start source:
name = pencil_beam # This name can be anything you like
library = egs_parallel_beam
charge = -1
direction = 0 0 1
:start spectrum:
type = monoenergetic
energy = 0.7 # in MeV
:stop spectrum:
:start shape:
type = point
position = 0 -0.5 0 # in cm
:stop shape:
:stop source:
### Use the source by this name for the simulation
simulation source = pencil_beam
:stop source definition:
################################
################################
:start view control:
# Here we are just assigning some colors for nice
# viewing in the egs_view application
set color = tungsten 120 255 120 255
:stop view control:
################################
################################
:start ausgab object definition: # Only 1 ausgab definition block allowed
### Particle tracks
:start ausgab object:
name = tracks
library = egs_track_scoring
:stop ausgab object:

###:start ausgab object:
###name = dose
###library = egs_dose_scoring
###dose regions = 1
###volume = 100
###:stop ausgab object:

:start ausgab object:
library                  = egs_phsp_scoring
name                     = backscattering1
output format            = EGSnrc
particle type            = all
score multiple crossers  = no
phase space geometry     = scoreplane1
:stop ausgab object:

:start ausgab object:
library                  = egs_phsp_scoring
name                     = backscattering2
output format            = EGSnrc
particle type            = all
score multiple crossers  = no
phase space geometry     = scoreplane2
:stop ausgab object:

:stop ausgab object definition:

I think it comes down to the question of how to generate a rectangular plane of a particular size, position. I checked the relevant information and couldn't find a clear order, so can you help me?

**Comment by CQUswl** [comment]
I recorded “scoreplane2” as above, and when I processed it with beamdp_gui, I found that because the recording surface is perpendicular to the Y-axis, it is different from the recording plane perpendicular to the Z-axis. Perpendicular to the Y axis, the range of the X and Z axes should be defined, but only the X and Y axes can be entered in beamdp_gui. So I can't determine the X and Y ranges in beamdp_gui, is there a clear correspondence? Also, I tried to give X a range from -1 cm to 1 cm. And the Y range is -1 cm to 1 cm, and the electrons in that range are read, and there is no electronic position output in the results.

**Reply by rtownson** [reply]
Phase-spaces in .egsphsp format are always assumed to be in the x-y plane. Output a phase-space in .IAEAphsp format instead, so that 3D data is allowed.
